### PR TITLE
CLI - Migration delete

### PR DESCRIPTION
**Pull request by v-harihar**
**Describe what this PR does**
1. Removes the ReplicationDestination that is associated w/
       the relationship (if it exists in-cluster)
2. Deletes the relationship config file

**Related issues:**
#83

### CODE REVIEW

File: kubectl-volsync/cmd/migration_delete.go [hunk 1]
```diff
@@ -17,36 +17,107 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
...
+	err = md.deleteReplicationDestination(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Wait for ReplicationDestination status to be Terminated
+	err = md.mr.data.Destination.waitForRDToTerminate(ctx, md.client)
+	if err != nil {
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (v-harihar) [reply kubectl-volsync/cmd/migration_delete.go:76]: This is to make sure `replicationdestination` is Terminated and objects handled by it are cleaned. 
Shall I remove it of not reqd ?

File: kubectl-volsync/cmd/migration_rsync.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ import (
 type migrationSync struct {
 	mr *migrationRelationship
 	// Address is the remote address to connect to for replication.
-	DestAddr string
+	DestAddr *string
```
[the remark this answers is not in the document]
Comment (v-harihar) [reply kubectl-volsync/cmd/migration_rsync.go:39]: Sorry.. I was making some changes and committed it by mistake. I will remove it

File: kubectl-volsync/cmd/migration_delete.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (JohnStrunk) [reply kubectl-volsync/cmd/migration_delete.go:76]: I would. Once the delete has been issued, the DeletionTimestamp has been set, and the object is destined for garbage collection.

[the remark this answers is not in the document]
Comment (v-harihar) [reply kubectl-volsync/cmd/migration_delete.go:76]: Makes sense

**Review by JohnStrunk — APPROVED** [review]
/lgtm

**Pull request merged**